flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -382,7 +381,10 @@ object UserDefinedFunctionUtils {
       implicitResultType: TypeInformation[_],
       params: Expression*): TableFunctionCall = {
     val arguments = transformLiteralExpressions(params: _*)
-    val userDefinedResultType = tableFunction.getResultType(arguments)
+    val typeInformations = params.map { param =>
+      if (param.valid) param.resultType else null
```

Comment:
Actually. The `param.valid` here is to skip the `UnresolvedFieldReference`.
For example.
```
    val result = in
      .join(funcDyn('c, 1) as 'name)
      .select('c, 'name)
```
will throw a `UnresolvedFieldReference`
```
org.apache.flink.table.api.UnresolvedException: Calling resultType on class org.apache.flink.table.expressions.UnresolvedFieldReference.

	at org.apache.flink.table.expressions.UnresolvedFieldReference.resultType(fieldExpression.scala:45)
	at org.apache.flink.table.functions.utils.UserDefinedFunctionUtils$$anonfun$10.apply(UserDefinedFunctionUtils.scala:385)
	at org.apache.flink.table.functions.utils.UserDefinedFunctionUtils$$anonfun$10.apply(UserDefinedFunctionUtils.scala:384)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:34)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
	at scala.collection.AbstractTraversable.map(Traversable.scala:105)
	at org.apache.flink.table.functions.utils.UserDefinedFunctionUtils$.buildTableFunctionCall(UserDefinedFunctionUtils.scala:384)
	at org.apache.flink.table.functions.TableFunction.apply(TableFunction.scala:91)
	at org.apache.flink.table.runtime.dataset.DataSetUserDefinedFunctionITCase.testDynamicSchema(DataSetUserDefinedFunctionITCase.scala:127)
```
Because `'c` is an `Expression` which has't been resolved.